dali/pipeline/pipeline_test.cc

```diff
@@ -311,113 +287,183 @@ TYPED_TEST(PipelineTest, TestSerialization) {
             original_graph.NumOp(OpType::GPU));
 }
 
-/*
-TYPED_TEST(PipelineTest, TestSinglePrefetchOp) {
-  int num_thread = TypeParam::nt;
-  int batch_size = this->jpegs_.size();
-  cudaStream_t stream;
-  CUDA_CALL(cudaStreamCreateWithFlags(&stream, cudaStreamNonBlocking));
-
-  Pipeline pipe(batch_size, num_thread, stream, 0, true);
-
-  // Add an external source
-
-  // Add a single prefetch op
-  pipe.AddTransform(OpSpec("Copy")
-      .AddArg("stage", "Prefetch"));
+class DummyPresizeOpCPU : public Operator<CPUBackend> {
+ public:
+  explicit DummyPresizeOpCPU(const OpSpec &spec)
+      : Operator<CPUBackend>(spec) {
+  }
 
-  // Build the pipeline
-  pipe.Build();
+  void RunImpl(Workspace<CPUBackend>* ws, int idx) override {
+    auto &input = ws->Input<CPUBackend>(idx);
+    auto &output = ws->Output<CPUBackend>(idx);
+    auto tmp_size = output.capacity();
+    output.mutable_data<size_t>();
+    output.Resize({2});
+    auto out = output.mutable_data<size_t>();
+    out[0] = tmp_size;
+    out[1] = input.capacity();
+  }
+};
 
-  // Run the pipeline
-  for (int i = 0; i < 5; ++i) {
-    pipe.RunPrefetch();
-    pipe.RunCopy();
-    pipe.RunForward();
+class DummyPresizeOpGPU : public Operator<GPUBackend> {
+ public:
+  explicit DummyPresizeOpGPU(const OpSpec &spec)
+      : Operator<GPUBackend>(spec) {
+  }
 
-    CUDA_CALL(cudaStreamSynchronize(stream));
+  void RunImpl(Workspace<GPUBackend>* ws, int idx) override {
+    auto &input = ws->Input<GPUBackend>(0);
+    auto &output = ws->Output<GPUBackend>(0);
+    output.mutable_data<size_t>();
+    size_t tmp_size[2] = {output.capacity(), input.capacity()};
+    std::vector< std::vector<Index> > shape;
+    shape.push_back(std::vector<Index>(1));
+    output.Resize(shape);
+    auto out = output.mutable_data<size_t>();
+    CUDA_CALL(cudaDeviceSynchronize());
+    CUDA_CALL(cudaMemcpy(out, &tmp_size, sizeof(size_t) * 2, cudaMemcpyDefault));
+  }
+};
 
-    // Verify the results
-    this->CompareData(
-        pipe.output_batch().template data<uint8>(),
-        batch->template data<uint8>(),
-        batch->size());
+class DummyPresizeOpMixed : public Operator<MixedBackend> {
+ public:
+  explicit DummyPresizeOpMixed(const OpSpec &spec)
+      : Operator<MixedBackend>(spec) {
   }
-}
 
-TYPED_TEST(PipelineTest, TestNoOps) {
-  int num_thread = TypeParam::nt;
+  using Operator<MixedBackend>::Run;
+  void Run(MixedWorkspace* ws) override {
+    auto &input = ws->Input<CPUBackend>(0, 0);
+    auto &output = ws->Output<GPUBackend>(0);
+    output.mutable_data<size_t>();
+    size_t tmp_size[2] = {output.capacity(), input.capacity()};
+    std::vector< std::vector<Index> > shape;
+    shape.push_back(std::vector<Index>(1));
+    output.Resize(shape);
+    auto out = output.mutable_data<size_t>();
+    CUDA_CALL(cudaDeviceSynchronize());
```

Comment:
Why not
```suggestion
    CUDA_CALL(cudaStreamSynchronize(ws->stream()));
```

Reply:
I don't know if we need to care in test for such optimization.

Reply:
Sure, I was just wondering if it has any meaning

Reply:
Anyway I have changed that too